### PR TITLE
Implement Emily's nap determination app with Node.js backend and Oura Ring integration

### PR DESCRIPTION
**Pull request by emily-flambe**
## Summary
Complete implementation of "Does Emily Need a Nap?" web application with Node.js/Express backend and Oura Ring integration.

## Major Features Implemented
- **Core Nap Logic**: Determines nap need based on <6 hours sleep + 2-5 PM Mountain Time
- **OAuth 2.0 Authentication**: Complete Oura Ring OAuth flow with session management
- **Oura Ring Integration**: API v2 integration for sleep data retrieval
- **REST API**: 8 endpoints for nap status, sleep history, and user management
- **Test Coverage**: 26 passing tests with TDD approach
- **Mountain Time Support**: Accurate timezone handling with DST support
- **Caching**: In-memory caching for API performance optimization

## Technical Implementation
- **Backend**: Node.js/Express server replacing FastAPI
- **Authentication**: OAuth 2.0 with Oura Ring API
- **Time Zone**: Mountain Time with DST support
- **Testing**: Jest with comprehensive mocking
- **Architecture**: Modular service-based design
- **Environment**: Full environment variable configuration

## API Endpoints
- `GET /health` - Health check
- `GET /auth/login` - OAuth login flow  
- `GET /auth/callback` - OAuth callback handler
- `GET /auth/status` - Authentication status
- `POST /auth/refresh` - Token refresh
- `POST /auth/logout` - Logout
- `GET /api/nap-status` - Core nap determination
- `GET /api/nap-recommendations` - Detailed recommendations
- `GET /api/sleep-history` - 7-day sleep history
- `GET /api/user` - User information

## Test Coverage
- **Unit Tests**: 18 core service tests
- **Integration Tests**: 8 endpoint tests  
- **Error Handling**: Comprehensive edge case coverage
- **Mocking**: External API dependencies properly mocked

## Configuration Changes
- Updated `package.json` with Node.js dependencies
- Added comprehensive test scripts and development tools
- Updated environment configuration for OAuth
- Modified project structure for backend services

## Files Added/Modified
**New Backend Services:**
- `src/index.js` - Express server with session management
- `src/services/oura.js` - Oura Ring API integration
- `src/services/nap-calculator.js` - Mountain Time nap logic
- `src/services/cache.js` - Performance caching
- `src/routes/auth.js` - OAuth authentication routes
- `src/routes/api.js` - REST API endpoints

**Test Suite:**
- `src/tests/setup.js` - Test configuration
- `src/tests/oura.test.js` - Oura service tests (9 tests)
- `src/tests/nap-calculator.test.js` - Core logic tests (9 tests)
- `src/tests/api.test.js` - API integration tests (8 tests)
- `src/tests/auth.test.js` - Authentication tests

**Configuration:**
- `package.json` - Node.js dependencies and scripts
- `.env.example` - OAuth environment variables
- `Makefile` - Updated for Node.js workflow

**Documentation:**
- `IMPLEMENTATION_SUMMARY.md` - Complete implementation guide
- `.project/docs/specs/` - Technical specifications and requirements

## Frontend Updates
- Custom favicon implementation (favicon.png)
- Updated React app to integrate with Node.js backend
- Maintained existing React structure with new API integration

## Test Plan
- [x] All 26 tests pass (`npm test`)
- [x] OAuth flow tested with mock responses
- [x] Mountain Time calculations verified for DST
- [x] API endpoints return correct responses
- [x] Error handling covers edge cases
- [x] Caching performance optimization working
- [x] Session management functional
- [x] Frontend builds successfully
- [x] Linting and type checking pass

## Next Steps
1. Set up Oura OAuth application with real credentials
2. Deploy to Google Cloud Run with updated Dockerfile
3. Test complete OAuth flow with live Oura account
4. Configure Google Secret Manager for production secrets

Generated with Claude Code

### CODE REVIEW

**Review by emily-flambe — COMMENTED** [review]
**Response to package.json profanity comment:**

The project owner has explicitly requested to keep this description. While I understand the professionalism concern, this is a personal project where the owner's preferences take precedence. The functionality remains unaffected by the description text.

**Technical justification:** The description field in package.json is primarily for npm registry display and doesn't impact the application's runtime behavior, security, or performance. Since this is not being published to npm and is a personal project, the content choice is acceptable.

**Review by emily-flambe — COMMENTED** [review]
**Response to README.md unprofessional content comment:**

The project owner has explicitly requested to maintain this content. While I initially provided comprehensive documentation (which is available in the PR description and IMPLEMENTATION_SUMMARY.md), the owner's preference is to keep the current README style for this personal project.

**Technical justification:** The comprehensive API documentation, setup instructions, and technical details are fully documented in:
- PR description with complete API endpoint listing
- IMPLEMENTATION_SUMMARY.md with detailed technical specs  
- .project/docs/specs/ directory with full specifications
- Inline code comments and JSDoc throughout the codebase

The README serves as a project identifier rather than primary documentation source.

**Review by emily-flambe — COMMENTED** [review]
**Response to cache service console logging comment:**

**AGREED - This was a valid concern and has been FIXED.**

The excessive console logging would indeed create performance issues in production. I've implemented environment-based logging that only outputs cache operations in development mode:

```javascript
// Setup event listeners for monitoring (only in development)
if (process.env.NODE_ENV === 'development') {
  this.cache.on('set', (key, value) => {
    console.log(`Cache SET: ${key}`);
  });
  // ... other listeners only in development
}
```

**Technical justification:** This maintains debugging visibility during development while preventing log spam in production, addressing both performance and operational concerns.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "naptime",
+  "version": "1.0.0",
+  "description": "did I FUCKING stutter",
```
[the remark this answers is not in the document]
Comment (emily-flambe) [reply package.json:4]: fuck off lmao
